pkg/skaffold/build/cluster/kaniko.go

```diff
@@ -133,7 +134,7 @@ func args(artifact *latest.KanikoArtifact, context, tag string, insecureRegistri
 			args = append(args, "--cache-repo", artifact.Cache.Repo)
 		}
 		if artifact.Cache.HostPath != "" {
-			args = append(args, "--cache-dir", artifact.Cache.HostPath)
+			args = append(args, "--cache-dir", constants.DefaultKanikoCacheDirMountPath)
```

Comment:
I don't understand this change here - the logic is - if the user overrode the cache path - we should use it, and you are changing it to "if the user overrode the cache path, ignore it and keep it the default"

Reply:
Hi,
I think there's a misunderstanding of this variable usage, IMHO, the "hostPath" variable should represent the path on the host where the kaniko pod is running. So the cache-dir kaniko's command represent the path inside the pod, this one doesn't really need to be modified.
As mentioned in the issue #3156 , with the current usage of the hostpath variable, kaniko is unable to find its cache; the current resulting kaniko's yaml file is as follow:

```yaml
containers:
  - args:
    ...
    - --cache-dir
    - /data/docker/kaniko/cache
    ...
    volumeMounts:
    - mountPath: /cache
      name: kaniko-cache
  ...
volumes:
  - hostPath:
      path: /data/docker/kaniko/cache
      type: ""
    name: kaniko-cache
```

As shown, the cache is mounted at "/cache" by kubernetes but search at "$hostPath" (/data/docker/kaniko/cache here) by kaniko. This is the reason of the patch. I have compiled skaffold with this patch and kaniko finds its cache perfectly.
Thank you for your time

Reply:
That makes sense to me. Assuming the cache exists on the node, users can specify exactly where it is via `hostPath` and then we can automatically mount it to some default location (I would use `/kaniko/<some dir>` as the mount point in the container.

Reply:
Hi,
The "/cache" is the default for kaniko and you defined the constant I use here to "/cache". So, I can also update the constant to be "/kaniko/cache" but I think it's less confusing to let at "/cache".

Reply:
Gotcha, thanks for explaining - I also read the docs :), it makes sense now.